File: Core CS/Algorithms/Warmup/Time Conversion/TimeConversion.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution {

    static string timeConversion(string s) {
        if(s[8]=='A')
            if(s.Substring(0,2) == "12") return "00" + s.Substring(2,6);
            else return s.Substring(0,8);
        else
            if(s.Substring(0,2) == "12") return s.Substring(0,8);
            else return ((Convert.ToInt32(s.Substring(0,2)) + 12)%24).ToString() + s.Substring(2,6);
    }

    static void Main(String[] args) {
        string s = Console.ReadLine();
        string result = timeConversion(s);
        Console.WriteLine(result);
    }
}

File: Core CS/Algorithms/Warmup/Time Conversion/timeConversion.scala
import java.io.PrintWriter

object Solution {

    def timeConversion(s: String): String = {
        if(s(8) == 'P')
            if( s.substring(0,2) == "12")
                return s.substring(0,8)
            else
                return (s.substring(0,2).toInt + 12).toString + s.substring(2,8)
        else
            if(s.substring(0,2) == "12")
                return "00" + s.substring(2,8)
            else
                return s.substring(0,8)
    }

    def main(args: Array[String]) {
        val scan = scala.io.StdIn

        val fw = new PrintWriter(sys.env("OUTPUT_PATH"))

        val s = scan.readLine

        val result = timeConversion(s)

        fw.println(result)

        fw.close()
    }
}
